src/Core/Employee/Access/ProfileAccessCheckerInterface.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Employee\Access;
+
+/**
+ * Interface ProfileAccessCheckerInterface describes a profile access checker.
+ */
+interface ProfileAccessCheckerInterface
+{
+    /**
+     * Checks if context employee can access given profile.
+     *
+     * @param int $profileId
+     *
+     * @return bool
+     */
+    public function canAccessProfile($profileId);
```

Comment:
it seems that `employeeId` would be useful here, it raises question: who do you check access against? you are forced to depend on context with this signature.

However, i'm not sure how easy it would be to implement that. @matks wdyt?